src/libraries/Microsoft.PowerFx.Interpreter/CustomFunction/ReflectionFunction.cs

```diff
@@ -156,21 +158,27 @@ internal TexlFunction GetTexlFunction()
                 };
             }
 
-            return new CustomTexlFunction(info.Name, info.RetType, info.ParamTypes)
+            return new CustomTexlFunction(info.Name, FunctionCategories.UserDefined, info.RetType, info.ParamTypes)
             {
                 _impl = (runtimeConfig, args, cancellationToken) => InvokeAsync(runtimeConfig, args, cancellationToken),
                 LamdaParamMask = info.LamdaParamMask,
             };
         }
 
+        internal string GetFunctionName()
+        {
+            var info = Scan();

```

Comment:
> var info = Scan(); [](http://example.com/codeflow?start=11&length=19)

Why do we need to call Scan? Do we have this info already in _info?

Reply:
if _info is null, scan will find the info.